presto-main/src/main/java/com/facebook/presto/server/ServerConfig.java

```diff
@@ -94,6 +94,13 @@ public ServerConfig setCoordinator(boolean coordinator)
         return this;
     }
 
+    public boolean isWorker()
+    {
+        // At this moment we use the inverted logic to determine a worker, but this is fragile since the presto architecture might add new components in the future.
+        // It would be better to have a flag set from the configuration.
+        return !isCoordinator() && !isCatalogServer() && !isResourceManager();
```

Comment:
Yeah, this is fragile. Why not introduce the flag now?

Reply:
By doing that we need to deploy every clusters to add the flag which is non-trivial works.